apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/RequestOptions.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+using System;
+
+namespace Google.Cloud.Spanner.Data
+{
+    /// <summary>
+    /// Immutable class representing request options.
+    /// </summary>
+    internal class RequestOptions : IEquatable<RequestOptions>
+    {
+        /// <summary>
+        /// The request tag configured in the options.
+        /// </summary>
+        internal string RequestTag { get => Proto.RequestTag; }
+
+        /// <summary>
+        /// The transaction tag configured in the options.
+        /// </summary>
+        internal string TransactionTag { get => Proto.TransactionTag; }
+
+        /// <summary>
+        /// Clones the options and sets the request tag to the given value.
+        /// </summary>
+        /// <returns>
+        /// A clone of the options with the updated request tag.
+        /// </returns>
+        /// <param name="tag">Request tag to set.</param>
+        internal RequestOptions WithRequestTag(string tag)
+        {
+           var protoCopy = Proto.Clone();
+           protoCopy.RequestTag = tag ?? "";
+           return new RequestOptions(protoCopy);
+        }
+
+        /// <summary>
+        /// Clones the options and sets the transaction tag to the given value.
+        /// </summary>
+        /// <returns>
+        /// A clone of the options with the updated transaction tag.
+        /// </returns>
+        /// <param name="tag">Transaction tag to set.</param>
+        internal RequestOptions WithTransactionTag(string tag)
+        {
+            var protoCopy = Proto.Clone();
+            protoCopy.TransactionTag = tag ?? "";
+            return new RequestOptions(protoCopy);
+        }
+
+        /// <summary>
+        /// The proto representation of the request options. Must not be mutated
+        /// or exposed publicly.
+        /// </summary>
+        internal V1.RequestOptions Proto { get; }
```

Comment:
Does this need to be internal rather than private, given that we have the ToProto method?